example/reader_util.c

```diff
@@ -1754,6 +1754,138 @@ void update_tcp_flags_count(struct ndpi_flow_info* flow, struct ndpi_tcphdr* tcp
 
 /* ****************************************************** */
 
+#define MIN_ENTROPY 2.5
+#define MAX_ENTROPY 4.8
+/**
+ * @brief Calculates the Shannon entropy of a given string to estimate its randomness.
+ *
+ * This function computes the Shannon entropy of the input string by analyzing the
+ * frequency distribution of its characters. Higher entropy indicates more randomness,
+ * while lower entropy suggests more predictable or structured content.
+ *
+ * The function is useful for detecting potentially random or pseudo-random strings,
+ * such as passwords, tokens, or hashes.
+ *
+ * @param str A null-terminated input string to analyze.
+ * @return The calculated entropy as a double. Returns 0.0 for empty strings.
+ *
+ * @note Typical entropy values range between MIN_ENTROPY (2.5) and MAX_ENTROPY (4.8)
+ *       for short strings. These thresholds can be used to classify the randomness
+ *       level of the string.
+ */
+static double calculate_entropy(const char *str) {
```

Comment:
no need to re-implement, already present in the library; see `ndpi_entropy()`

Reply:
+1